src/coprocessor/codec/data_type/vector.rs

```diff
@@ -443,6 +443,160 @@ impl VectorValue {
 
         Ok(())
     }
+
+    /// Returns maximum encoded size in binary format.
+    pub fn maximum_encoded_size(&self) -> Result<usize> {
+        match self {
+            VectorValue::Int(ref vec) => Ok(vec.len() * 9),
+
+            // Some elements might be NULLs which encoded size is 1 byte. However it's fine because
```

Comment:
Will it be a problem if there are almost NULLs?

Reply:
No. It will just allocate more memory than really used, but will not cause any problems.